src/concepts/components/internalstate.md

```diff
@@ -0,0 +1,169 @@
+---
+description: Component can maintain it's own state and render information depending on it
+---
+
+# Internal State
+
+The component can manage it's own state using the Rust struct that implement the trait `Component`. The HTML rendering is based on this state.
+When state change there is possibility to re-render the component.
```

Comment:
"wehn state" => "when the state"
"there is possibility ot re-render the component" => "the component might be re-rendered"